lib/command/definitions.js

```diff
@@ -169,75 +70,44 @@ module.exports = function (genPath, options) {
-    const pageObject = supports[name];
-    const pageMethods = addAllMethodsInObject(pageObject, {}, []);
-    let pageObjectExport = pageObjectTemplate.replace('{{methods}}', pageMethods.join(''));
-    pageObjectExport = pageObjectExport.replace('{{name}}', name);
-    exportPageObjects.push(pageObjectExport);
+  for (const name in codecept.config.include) {
+    const includePath = codecept.config.include[name];
+    if (includePath) customFilePaths[name] = includePath;
+    if (name === 'I' || name === translations.I) continue;
+    callbackParams.push(`arg_${name}:typeof ${name}`);
   }
 
-  let definitionsTemplate = template.replace('{{methods}}', methods.join(''));
-  definitionsTemplate = definitionsTemplate.replace('{{exportPageObjects}}', exportPageObjects.join('\n'));
-  definitionsTemplate = definitionsTemplate.replace('{{callbackParams}}', `, ${[...callbackParams, '...args: any'].join(', ')}`);
-  if (translations) {
-    definitionsTemplate = definitionsTemplate.replace(/\{\{I\}\}/g, translations.I);
-  }
+  const importPaths = getImportString(testsPath, targetFolderPath);
+
+  const definitionsTemplate = template({
+    helperNames: Object.keys(helpers),
+    callbackParams: [undefined, ...callbackParams, '...args: any[]'].join(', '),
+    importPaths,
+    translations,
+  });
 
   fs.writeFileSync(path.join(targetFolderPath, 'steps.d.ts'), definitionsTemplate);
   output.print('TypeScript Definitions provide autocompletion in Visual Studio Code and other IDEs');
   output.print('Definitions were generated in steps.d.ts');
 };
 
-function addAllMethodsInObject(supportObj, actions, methods, translations) {
-  for (const action of methodsOfObject(supportObj)) {
-    const fn = supportObj[action];
-    if (!fn.name) {
-      Object.defineProperty(fn, 'name', { value: action });
-    }
-    const actionAlias = translations ? translations.actionAliasFor(action) : action;
-    if (!actions[actionAlias]) {
-      methods.push(toTypeDef(fn));
-      actions[actionAlias] = 1;
-    }
-  }
-  return methods;
-}
+function getImportString(testsPath, targetFolderPath) {
+  const importStrings = [];
 
-function addAllNamesInObject(supportObj, actions, methods) {
-  for (const name in supportObj) {
-    if (actions[name]) {
-      continue;
-    }
-    const actor = supportObj[name];
-    let params = toTypeDef(actor);
-    if (params !== undefined) {
-      if (params.indexOf(' : ') > 0) {
-        if (params.indexOf(' (') > 0) {
-          params = params.trim();
-          methods.push(`    ${(name)}${params}\n`);
-        } else {
-          methods.push(`${params}`);
-        }
-      } else {
-        methods.push(`    ${(name)}: (${params}) => any; \n`);
-      }
-    }
+  for (const name in customFilePaths) {
+    const value = customFilePaths[name];
+    const parsedPath = path.parse(value);
+    const relativePath = path.relative(targetFolderPath, path.join(testsPath, parsedPath.dir, parsedPath.name));
```

Comment:
We can use PageObject from custom package. And then we get an error, like:
```
import RadioGroup = require('./../../tests/codeceptjs/configs/@custom_package/codeceptjs/PageObjects/Fields/Core/RadioGroup');
```

but it has to generate so:
```
require('@custom_package/codeceptjs/PageObjects/Fields/Core/RadioGroup');
```

I suggest you to look in container.createHelpers function